Mathlib/CategoryTheory/Enriched/Linear.lean

```diff
@@ -0,0 +1,137 @@
+  id X := ModuleCat.ofHom <| (LinearMap.ringLmapEquivSelf R R (X ⟶ X)).symm (𝟙 X)
+  comp X Y Z := ModuleCat.ofHom <| lift (Linear.comp X Y Z)
+  id_comp X Y := by
+    ext f
+    simp only [LinearMap.ringLmapEquivSelf_symm_apply, ModuleCat.hom_comp,
+      ModuleCat.MonoidalCategory.carrier_of_tensorObj_of, hom_ofHom,
+      LinearMap.coe_comp, Function.comp_apply, ModuleCat.hom_id, LinearMap.id_coe, id_eq] at f ⊢
+    rw [ModuleCat.MonoidalCategory.leftUnitor_inv_hom_apply]
+    simp [ModuleCat.MonoidalCategory.tensorUnit_eq]
+  comp_id X Y := by
+    ext f
+    simp only [LinearMap.ringLmapEquivSelf_symm_apply, ModuleCat.hom_comp,
+      ModuleCat.MonoidalCategory.carrier_of_tensorObj_of, hom_ofHom,
+      LinearMap.coe_comp, Function.comp_apply, ModuleCat.hom_id, LinearMap.id_coe, id_eq] at f ⊢
+    rw [ModuleCat.MonoidalCategory.rightUnitor_inv_hom_apply]
+    simp [ModuleCat.MonoidalCategory.tensorUnit_eq]
+  assoc W X Y Z := by
+    ext f
+    change _ ⊗[R] _ ⊗[R] _ at f
+    simp only [ModuleCat.hom_comp, ModuleCat.MonoidalCategory.carrier_of_tensorObj_of, hom_ofHom,
+      LinearMap.coe_comp, Function.comp_apply] at f ⊢
+    erw [lift_comp_lift_comp_rTensor_eq]
+    congr
+    exact (TensorProduct.assoc R (W ⟶ X) (X ⟶ Y) (Y ⟶ Z)).right_inv f
+  homEquiv {X Y} := (ModuleCat.homEquiv.trans
+      (LinearMap.ringLmapEquivSelf R R (X ⟶ Y)).toEquiv).symm
+  homEquiv_id X := rfl
+  homEquiv_comp {X Y Z} f g := by
+    ext
+    change (LinearMap.toSpanSingleton R (X ⟶ Z) (f ≫ g)) 1 =
+      (lift (Linear.comp X Y Z))
+        ((ModuleCat.Hom.hom
+          (ModuleCat.ofHom (LinearMap.toSpanSingleton R (X ⟶ Y) f) ⊗
+            ModuleCat.ofHom (LinearMap.toSpanSingleton R (Y ⟶ Z) g)))
+          (1 ⊗ₜ 1))
+    simp
+
+end EnrichedOfLinear
+
+noncomputable section LinearOfEnriched
+
+variable {R : Type u} [CommRing R]
+variable {C : Type u} [Category.{u} C] [EnrichedOrdinaryCategory (ModuleCat R) C]
+
+variable (R)
+
+/-- The commutative group structure on morphisms of a category `C` enriched over `ModuleCat R`. -/
+abbrev addCommGroupEnrichedModuleCatHom (X Y : C) : AddCommGroup (X ⟶ Y) :=
+  (eHomEquiv (V := ModuleCat R)).addCommGroup
+
+instance moduleModuleCatHom {X Y : C} :
+    letI : AddCommGroup (X ⟶ Y) := addCommGroupEnrichedModuleCatHom R X Y
+    Module R (X ⟶ Y) :=
+  EnrichedOrdinaryCategory.homEquiv.module R
+
+/-- The preadditive structure on `C` induced by `C` being enriched over `ModuleCat R` for a
+commutative ring `R`. -/
+abbrev preadditiveEnrichedModuleCat : Preadditive C where
+  homGroup X Y := addCommGroupEnrichedModuleCatHom R X Y
+  add_comp X Y Z f f' g := by
+    apply (eHomEquiv (ModuleCat R)).injective
+    simp [Equiv.add_def, eHomEquiv_comp, MonoidalPreadditive.add_tensor]
+  comp_add X Y Z f g g' := by
+    apply (eHomEquiv (ModuleCat R)).injective
+    simp [Equiv.add_def, eHomEquiv_comp, MonoidalPreadditive.tensor_add]
+
+variable {R}
+
+example : MonoidalLinear R (ModuleCat R) := inferInstance
```

Comment:
```suggestion
```
... although if you prefer to keep this example around as a sanity check, that's fine as well!